featurebyte/routes/target_table/controller.py

```diff
@@ -0,0 +1,151 @@
+"""
+Target API route controller
+"""
+from __future__ import annotations
+
+from typing import Optional
+
+from http import HTTPStatus
+
+from bson import ObjectId
+from fastapi import HTTPException, UploadFile
+
+from featurebyte.common.utils import dataframe_from_arrow_stream
+from featurebyte.models.target_table import TargetTableModel
+from featurebyte.routes.common.base_materialized_table import BaseMaterializedTableController
+from featurebyte.routes.task.controller import TaskController
+from featurebyte.schema.info import TargetTableInfo
+from featurebyte.schema.target_table import TargetTableCreate, TargetTableList
+from featurebyte.schema.task import Task
+from featurebyte.service.entity_validation import EntityValidationService
+from featurebyte.service.feature_store import FeatureStoreService
+from featurebyte.service.observation_table import ObservationTableService
+from featurebyte.service.preview import PreviewService
+from featurebyte.service.target import TargetService
+from featurebyte.service.target_table import TargetTableService
+from featurebyte.storage import Storage
+
+
+class TargetTableController(
+    BaseMaterializedTableController[TargetTableModel, TargetTableService, TargetTableList],
+):
+    """
+    TargetTable Controller
+    """
+
+    paginated_document_class = TargetTableList
+
+    def __init__(
+        self,
+        target_table_service: TargetTableService,
+        preview_service: PreviewService,
+        feature_store_service: FeatureStoreService,
+        observation_table_service: ObservationTableService,
+        entity_validation_service: EntityValidationService,
+        task_controller: TaskController,
+        target_service: TargetService,
+    ):
+        super().__init__(service=target_table_service, preview_service=preview_service)
+        self.feature_store_service = feature_store_service
+        self.observation_table_service = observation_table_service
+        self.entity_validation_service = entity_validation_service
+        self.task_controller = task_controller
+        self.target_service = target_service
+
+    async def create_target_table(
```

Comment:
this function is somewhat similar to its `historical_feature_table` counterpart. i'll refactor in a follow-up - https://featurebyte.atlassian.net/browse/DEV-1910

mostly copied over for now to get this up and running quickly.